test/unit/transaction-naming.test.js

```diff
@@ -5,150 +5,166 @@
-      expect(transaction.name).equal('WebTransaction/NormalizedUri/abc/*')
-      done()
+      t.equal(transaction.name, 'WebTransaction/NormalizedUri/test-transaction')
+      t.equal(transaction.name, transaction.getFullName(), 'name should be equal to finalized name')
+      t.end()
     })
   })
 
-  it('Custom naming rules should be cleaned up', function (done) {
+  t.test(
+    'Server sent naming rules should be applied when user specified rules are set',
+    function (t) {
+      agent.urlNormalizer.addSimple(/\d+/, '*')
+      agent.userNormalizer.addSimple(/123/, 'abc')
+      helper.runInTransaction(agent, function (transaction) {
+        transaction.finalizeNameFromUri('http://test.test.com/123/456', 200)
+        t.equal(transaction.name, 'WebTransaction/NormalizedUri/abc/*')
+        t.equal(
+          transaction.name,
+          transaction.getFullName(),
+          'name should be equal to finalized name'
+        )
+        t.end()
+      })
+    }
+  )
+
+  t.test('Custom naming rules should be cleaned up', function (t) {
     agent.userNormalizer.addSimple(/\//, 'test-transaction')
     helper.runInTransaction(agent, function (transaction) {
       transaction.finalizeNameFromUri('http://test.test.com/', 200)
-      expect(transaction.name).equal('WebTransaction/NormalizedUri/test-transaction')
-      done()
+      t.equal(transaction.name, 'WebTransaction/NormalizedUri/test-transaction')
+      t.equal(transaction.name, transaction.getFullName(), 'name should be equal to finalized name')
+      t.end()
     })
   })
 
-  it('Custom naming rules should trump instrumentation naming', function (done) {
+  t.test('Custom naming rules should trump instrumentation naming', function (t) {
     agent.userNormalizer.addSimple(/\//, '/test-transaction')
     helper.runInTransaction(agent, function (transaction) {
       simulateInstrumentation(transaction)
       transaction.finalizeNameFromUri('http://test.test.com/', 200)
-      expect(transaction.name).equal('WebTransaction/NormalizedUri/test-transaction')
-      done()
+      t.equal(transaction.name, 'WebTransaction/NormalizedUri/test-transaction')
+      t.equal(transaction.name, transaction.getFullName(), 'name should be equal to finalized name')
+      t.end()
     })
   })
 
-  it('Custom naming rules should trump API calls', function (done) {
+  t.test('API calls should trump Custom naming rules', function (t) {
     agent.userNormalizer.addSimple(/\//, '/test-transaction')
     const api = new API(agent)
     helper.runInTransaction(agent, function (transaction) {
       api.setTransactionName('override')
       transaction.finalizeNameFromUri('http://test.test.com/', 200)
-      expect(transaction.name).equal('WebTransaction/NormalizedUri/test-transaction')
-      done()
+      t.equal(transaction.name, 'WebTransaction/Custom/override')
+      t.equal(transaction.name, transaction.getFullName(), 'name should be equal to finalized name')
```

Comment:
as mentioned in PR details. if you put this line in main and run this suite, it'd fail, hence where the bug laid